selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/WorkspacesListTest.java

```diff
@@ -0,0 +1,377 @@
+
+import com.google.inject.Inject;
+import java.net.URL;
+import java.nio.file.Paths;
+import java.util.List;
+import org.eclipse.che.selenium.core.SeleniumWebDriver;
+import org.eclipse.che.selenium.core.TestGroup;
+import org.eclipse.che.selenium.core.client.TestProjectServiceClient;
+import org.eclipse.che.selenium.core.client.TestWorkspaceServiceClient;
+import org.eclipse.che.selenium.core.user.DefaultTestUser;
+import org.eclipse.che.selenium.core.webdriver.SeleniumWebDriverHelper;
+import org.eclipse.che.selenium.core.workspace.InjectTestWorkspace;
+import org.eclipse.che.selenium.core.workspace.TestWorkspace;
+import org.eclipse.che.selenium.pageobject.dashboard.Dashboard;
+import org.eclipse.che.selenium.pageobject.dashboard.DocumentationPage;
+import org.eclipse.che.selenium.pageobject.dashboard.NewWorkspace;
+import org.eclipse.che.selenium.pageobject.dashboard.workspaces.*;
+import org.eclipse.che.selenium.pageobject.dashboard.workspaces.Workspaces.Statuses;
+import org.testng.annotations.*;
+
+/**
+ * @author Sergey Skorik
+ * @author Ihor Okhrimenko
+ */
+@Test(groups = TestGroup.OSIO)
+public class WorkspacesListTest {
+  private static final int EXPECTED_WORKSPACES_COUNT = 2;
+  private static final int BLANK_WS_MB = 2048;
+  private static final int JAVA_WS_MB = 3072;
+  private static final int BLANK_WS_PROJECTS_COUNT = 0;
+  private static final int JAVA_WS_PROJECTS_COUNT = 1;
+  private static final String EXPECTED_DOCUMENTATION_PAGE_TITLE = "What Is a Che Workspace?";
+  private static final String EXPECTED_JAVA_PROJECT_NAME = "web-java-spring";
+  private static final String NEWEST_CREATED_WORKSPACE_NAME = "just-created-workspace";
+
+  @Inject private Dashboard dashboard;
+  @Inject private WorkspaceProjects workspaceProjects;
+  @Inject private WorkspaceConfig workspaceConfig;
+  @Inject private DefaultTestUser defaultTestUser;
+  @Inject private Workspaces workspaces;
+  @Inject private NewWorkspace newWorkspace;
+  @Inject private TestProjectServiceClient testProjectServiceClient;
+  @Inject private TestWorkspaceServiceClient testWorkspaceServiceClient;
+  @Inject private SeleniumWebDriverHelper seleniumWebDriverHelper;
+  @Inject private SeleniumWebDriver seleniumWebDriver;
+  @Inject private DocumentationPage documentationPage;
+  @Inject private WorkspaceOverview workspaceOverview;
+
+  @InjectTestWorkspace(template = DEFAULT, memoryGb = 2, startAfterCreation = false)
+  private TestWorkspace blankWorkspace;
+
+  @InjectTestWorkspace(template = UBUNTU_JDK8, memoryGb = 3)
+  private TestWorkspace javaWorkspace;
+
+  private Workspaces.WorkspaceListItem expectedBlankItem;
+  private Workspaces.WorkspaceListItem expectedJavaItem;
+  private Workspaces.WorkspaceListItem expectedNewestWorkspaceItem;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    URL resource = getClass().getResource("/projects/defaultSpringProjectWithDifferentTypeOfFiles");
+
+    blankWorkspace.await();
```

Comment:
Next commands are redundant because we waiting for those workspaces during injection.
```
blankWorkspace.await()
javaWorkspace.await();
```